docs/design/log_tracing.md

```diff
@@ -0,0 +1,137 @@
+# Logging and Tracing
+
+A few ideas around logging and tracing.
+
+## TL;DR
+
+A quick summary:
+
+* Don't panic, don't println (see below for exceptions).
+* [Log messages](https://crates.io/crates/log) and [tracing](https://crates.io/crates/tracing) should be relevant to
+  operating personell and developers, not end users. Ops will either run
+  with a level of `warn` or `info` when sending to a tracing solution (like Jaeger).
+* Errors and warnings, which are relevant to the end user, must be sent back using the same channel (API call, UI, …)
+  not the logging system
+* Be reasonable with the amount of data being logged (might be transmitted to a tracing system, like Jaeger)
+* Instrumentation (`#[instrument]` and friends)
+  * Only apply to time-consuming or complex functions
+  * Skip large fields: `#[instrument(skip(self, large_arg))]`
+  * Add return values (as info) if relevant: `#[instrument(ret)]`
+  * For larger return values, skip the "ok value" using `err` with a custom level: `#[instrument(err(level=tracing::Level::INFO))]`
+
+## Rationale
+
+Logging and tracing is intended for either developers or operators (humans) of the system. Not for end users.
+
+This means that errors or warnings relevant to the end user should be propagated through the API and UI to the user,
+not the console or logging/tracing systems like opentelemetry.
+
+## Levels
+
+* `error` – Something went seriously wrong, the application can no longer work properly and will most likely exit next.
+* `warning` – The application encountered a serious issue. However, it can continue to run and future operations are likely to succeed of the cause has been resolved.
```

Comment:
```suggestion
* `warning` – The application encountered a serious issue. However, it can continue to run and future operations are likely to succeed if the cause has been resolved.
```